sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,467 @@
+int vectorSum(float *v, size_t s, size_t sz) {
+  ...
+}
+
+{
+  sycl::queue q{...};
+
+  constexpr int kCount = ...;
+  std::vector<float> in(kCount);
+  float* out = ...;
+  ...
+
+  q.single_task([=] {
+    task_sequence<vectorSum> firstHalf;
+    task_sequence<vectorSum> secondHalf;
+    constexpr int halfCount = kCount/2;
+    firstHalf.async(in.data(), 0, halfCount);
+    secondHalf.async(in.data(), halfCount, halfCount);
+    *out = firstHalf.get() + secondHalf.get();
+  }).wait();
+  ...
+}
+```
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_FPGA_TASK_SEQUENCE` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to determine
+if the implementation supports this feature, or applications can test the
+macro's value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+== Definitions
+
+== `task_sequence` class
+
+[source,c++,linenums]
+----
+namespace sycl::ext::intel::experimental {
+
+template<auto &f> struct task_sequence_checker : std::false_type {};
+
+template<typename ReturnT, typename... ArgsT, ReturnT (&f)(ArgsT...)>
+struct task_sequence_checker<f> : std::true_type {};
+
+template<auto& f, class propertiesT =
+  decltype(oneapi::experimental::properties{})>
+class task_sequence{ static_assert(task_sequence_checker<f>::value); };
+
+template <typename ReturnT, typename... ArgsT, ReturnT (&f)(ArgsT...),
+          class propertiesT>
+class task_sequence<f, propertiesT> {
+
+public:
+  task_sequence(const task_sequence &) = delete;
+  task_sequence &operator=(const task_sequence &) = delete;
+  task_sequence(task_sequence &&) = delete;
+  task_sequence &operator=(task_sequence &&) = delete;
+
+  task_sequence();
+
+  task_sequence(const propertiesT &);
+
+  void async(ArgsT... args);
+
+  ReturnT get();
+
+  template <typename propertyT>
+  static constexpr bool has_property();
+
+  template <typename propertyT>
+  static constexpr auto get_property();
+
+  ~task_sequence();
+};
+
+} // namespace experimental
+} // namespace intel
+} // namespace ext
+} // namespace sycl
+
+----
+
+`task_sequence` is a class template, parameterized by an `auto` reference to a
++Callable+ `f` and by a list of properties `propertiesT`.
+The +Callable+ `f` defines the asynchronous task to be
+associated with the `task_sequence`, and requiring an auto reference ensures
+that each `f` is statically resolvable at compile time.  Static resolvability by
+the compiler is desirable when compiling for spatial architectures as it can
+enable the generation of more efficient hardware. Furthermore, the partial
+specialization:
```

Comment:
This doesn't make sense now; it's been incompletely updated after the latest changes.  The partial specialization of task_sequence _itself_ makes it easier to write the member functions, but the snippet below shows the checker.  The checker is used for validation.  

Also, I'm not sure that the checker and the partial specialization should even be shown.  Are they implementation details that should just be described in words?  E.g.: "templating a task_sequence on a non-callable is malformed - diagnostic required".  @gmlueck, what you do think?  How much of the metaprogramming should be explicit in the specification?

Reply:
The synopsis should not show implementation details.  If the checker is not part of the API, then don't show it.  You should describe requirements in words.

Reply:
I've removed the lines in the code block containing the checker

Reply:
I think we should go further and remove all references to the partial specialization and instead explain any restrictions on valid template parameters in words.

Reply:
I've removed the mention of partial specialization, wondering what you were thinking in addition to the point in the comment chain above as well as the existing paragraph in the doc that is needed here?

Reply:
I've removed the definitions for the partial specialization, and instead described the async, get, and destructor functionality in words using terms based on the Callable definition in the C++ spec.

Reply:
I think you've swung too far the other way now.  We should specify async and get as they are part of the user API and are therefore not implementation details.  Generally we would put them in the listing and in a table of member functions.  You can look at the section about accessor in the core spec for an example.